testnet-client/src/main.rs

```diff
@@ -358,23 +361,24 @@ string that we send you. It should look something like:
 
         // Read and parse B58 request code.
         #[derive(Clone)]
-        struct WrappedRequestPayload(pub Option<RequestPayload>);
+        struct WrappedRequestPayload(pub Option<PaymentRequest>);
         impl FromStr for WrappedRequestPayload {
             type Err = String;
             fn from_str(src: &str) -> Result<Self, Self::Err> {
                 if src.is_empty() {
                     return Ok(Self(None));
                 }
-
-                Ok(Self(Some(RequestPayload::decode(src).map_err(|err| {
-                    format!("Invalid request code: {}", err)
-                })?)))
+                let request_wrapper = PrintableWrapper::b58_decode(src.to_string())
+                    .map_err(|err| format!("Invalid request code: {}", err))?;
+                Ok(Self(Some(request_wrapper.get_payment_request().clone())))
             }
         }
         impl fmt::Display for WrappedRequestPayload {
             fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
                 if let Some(inner) = &self.0 {
-                    write!(f, "{}", inner.encode())?;
+                    let mut wrapper = PrintableWrapper::new();
+                    wrapper.set_payment_request(inner.clone());
+                    write!(f, "{:?}", wrapper.b58_encode())?;
```

Comment:
I think this should be `{}` otherwise the output would be wrapped in quotes which I don't think we want. Additionally, b58_encode() could fail, so this should be handled and propagated. I think `b58_encode().map_err(|_e| fmt::Error)?` should do it.